### PR TITLE
Update for current Stratosphere

### PR DESCRIPTION
**Pull request by thestr4ng3r**
I did not yet get it to work, but at least it compiles.
Feel free to review already if you want.

### CODE REVIEW

**Comment by jakibaki** [comment]
:+1: 

Just in case you didn't know you need to patch out the ams included hid-mitm out of atmosphere if you want my hid-mitm to work with current atmosphere rn.

Afaik with the next release that won't be included in atmosphere anymore but for now it's needed.

File: hid_mitm/source/hid_mitm_service.hpp [hunk 1]
```diff
@@ -17,53 +17,55 @@
...
     }
 
-    static void PostProcess(IMitmServiceObject *obj, IpcResponseContext *ctx);
-
   protected:
     /* Overridden commands. */
-    virtual Result CreateAppletResource(Out<std::shared_ptr<IAppletResourceMitmService>> out, PidDescriptor pid, u64 arid) final;
-    virtual Result ReloadConfig() final;
-    virtual Result ClearConfig() final;
+    virtual ams::Result CreateAppletResource(ams::sf::Out<std::shared_ptr<IAppletResourceMitmService>> out, ams::os::ProcessId pid, u64 arid) final;
```
Comment (HookedBehemoth) [inline hid_mitm/source/hid_mitm_service.hpp]: This has to be:
`ams::sf::Out<std::shared_ptr<IAppletResourceMitmService>> out, const sf::ClientAppletResourceUserId &client_aruid`

**Comment by HookedBehemoth** [comment]
Could you try my suggestion?

**Comment by thestr4ng3r** [comment]
I didn't have the time yet, but I plan to try it around the weekend. If you find anything else, feel free to add more comments.

File: hid_mitm/source/hid_mitm_service.hpp (hunk 1 above)
Comment (thestr4ng3r) [reply hid_mitm/source/hid_mitm_service.hpp]: Are you sure about that? This says `Takes a PID and an u64 AppletResourceUserId. Returns an #IAppletResource`: https://switchbrew.org/wiki/HID_services#CreateAppletResource

Comment (HookedBehemoth) [reply hid_mitm/source/hid_mitm_service.hpp]: Yes I am.
Check AMS source for their hid-mitm mitigation.

**Comment by staringstar** [comment]
Is this working?

**Comment by thestr4ng3r** [comment]
Nope, still crashes for me. But I am working on it as well as appreciate any help.

**Comment by thestr4ng3r** [comment]
works now \o/

**Comment by HookedBehemoth** [comment]
Congrats

**Comment by HookedBehemoth** [comment]
You should probably go way lower on stack and heap size.

**Comment by thestr4ng3r** [comment]
I had crashes with the smaller stack size before, but yeah can probably be lower now.

Somehow it seems to work now on all titles I tried so far except Link's Awakening where on startup, the hid sysmodule crashes like this with 2162-0002 (svcBreak): 
[01582576898_0100000000000013.log](https://github.com/jakibaki/hid-mitm/files/4246791/01582576898_0100000000000013.log)
Any idea what in hid-mitm might cause this?

**Comment by jakibaki** [comment]
First of all thank you very much! \o/

No clue whatsoever what might cause the svcBreak, might actually be unrelated to your changes. (hid-mitm never was particularly stable software).

I'll merge this for now :)

**Pull request merged**